sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -309,7 +309,7 @@ namespace Azure { namespace Core { namespace Http {
      * @brief Get HTTP body as #Azure::Core::IO::BodyStream.
      *
      */
-    Azure::Core::IO::BodyStream* GetBodyStream() { return this->m_bodyStream; }
+    Azure::Core::IO::BodyStream* GetBodyStream() const { return this->m_bodyStream; }
```

Comment:
Unrelated change - remove from this PR.

It's public surface area and **removing** a `const` would be a breaking change, so we need strong guidance on when we want to use `const` (because we are confident we'll never need to remove it), vs not having it.

Reply:
This *NOT* an unrelated change. It is required to get the code to correctly compile.

Reply:
@ahsonkhan Adding const to a method is not a breaking change.
Previously:
```cpp
void f(Azure::Core::Http::Request& req) {
    req.GetBodyStream(); // compiles
}

void g(Azure::Core::Http::Request const& req) {
    req.GetBodyStream(); // wouldn't compile
}
```

Now:
```cpp
void f(Azure::Core::Http::Request& req) {
    req.GetBodyStream(); // compiles
}

void g(Azure::Core::Http::Request const& req) {
    req.GetBodyStream(); // compiles
}
```

I have one suggestion though, I'll drop it in a separate comment.